maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java

```diff
@@ -183,19 +196,26 @@ private void multiThreadedProjectTaskSegmentBuild( ConcurrencyDependencyGraph an
     private Callable<ProjectSegment> createBuildCallable( final MavenSession rootSession,
                                                           final ProjectSegment projectBuild,
                                                           final ReactorContext reactorContext,
-                                                          final TaskSegment taskSegment, final ThreadOutputMuxer muxer )
+                                                          final TaskSegment taskSegment,
+                                                          final ThreadOutputMuxer muxer,
+                                                          final Set<String> duplicateArtifactIds )
     {
         return () ->
         {
             final Thread currentThread = Thread.currentThread();
             final String originalThreadName = currentThread.getName();
-            currentThread.setName( "mvn-builder-" + projectBuild.getProject().getId() );
+            final MavenProject project = projectBuild.getProject();
+
+            final String threadNameSuffix = duplicateArtifactIds.contains( project.getArtifactId() )
+                    ? project.getGroupId() + ":" + project.getArtifactId()
+                    : project.getArtifactId();
+            currentThread.setName( "mvn-builder_" + threadNameSuffix );
```

Comment:
Do you expect this to be a problem? This adds inconsistency to the rest of our thread names. I would only address it if is a problem only.

Reply:
I did it just for readability. It's not a big thing, so I can change it back.

Reply:
Please revert, we can discuss this when this is really going to be a problem.